### PR TITLE
Generate CodeView debug information for .asm files

### PR DESCRIPTION
**Pull request by AustinWise**
Specifically for Windows X64 and X86. ARM and ARM64 enable debug information using `-g` earlier in this file.

This enables seeing the original assembly source when using a debugger:

image

For reference:

* [Command line flags for X86 and X64 assembler](https://learn.microsoft.com/en-us/cpp/assembler/masm/ml-and-ml64-command-line-reference?view=msvc-170)
* [Command line flags for ARM and ARM64 assembler](https://learn.microsoft.com/en-us/cpp/assembler/arm/arm-assembler-command-line-reference?view=msvc-170)

### CODE REVIEW

File: eng/native/configurecompiler.cmake [hunk 1]
```diff
@@ -781,6 +781,8 @@ if (CLR_CMAKE_HOST_WIN32)
       set(CMAKE_ASM_MASM_COMPILE_OPTIONS_MSVC_RUNTIME_LIBRARY_MultiThreadedDLL      "")
       set(CMAKE_ASM_MASM_COMPILE_OPTIONS_MSVC_RUNTIME_LIBRARY_MultiThreadedDebug    "")
       set(CMAKE_ASM_MASM_COMPILE_OPTIONS_MSVC_RUNTIME_LIBRARY_MultiThreadedDebugDLL "")
+      # Generate CodeView debug information.
+      set(CMAKE_ASM_MASM_FLAGS "${CMAKE_ASM_MASM_FLAGS} /Zi")
```
Comment (jkoritzinsky) [inline eng/native/configurecompiler.cmake:785]: Can we use `add_compile_options` here instead of setting the `CMAKE_ASM_MASM_FLAGS` variable directly?

Comment (AustinWise) [reply eng/native/configurecompiler.cmake:785]: Earlier in the file `add_compile_options` is used for C and C++. I tried adding it there and tested that .ASM files and C++ files still have usable symbols.

https://github.com/dotnet/runtime/blob/960e4d723c27a5407dc691d06e546bc455a9c4a5/eng/native/configurecompiler.cmake#L673

**Comment by AustinWise** [comment]
@jkoritzinsky Does this way of turning on the `/Zi` flag look better to you?

**Review by jkoritzinsky — APPROVED** [review]
This way looks great to me!

**Comment by jkoritzinsky** [comment]
This failure is unrelated (this change only affects the Windows build)

**Pull request merged**